public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/SpanDetailLinkButtons.test.tsx

```diff
@@ -83,12 +92,71 @@ describe('getSpanDetailLinkButtons', () => {
       datasourceType: 'test',
       traceToProfilesOptions: undefined,
       timeRange,
+      app: CoreApp.Explore,
     });
 
     expect(result.logLinkButton).toBeNull();
     expect(result.profileLinkButtons).toBeNull();
     expect(result.sessionLinkButton).toBeDefined();
   });
+
+  it('should create profile drilldown button when plugin link exists', () => {
+    createSpanLink.mockReturnValue([{ type: SpanLinkType.Profiles, href: '/profiles', title: RelatedProfilesTitle }]);
+    (usePluginLinks as jest.Mock).mockReturnValue({
+      isLoading: false,
+      links: [{
+        pluginId: 'grafana-pyroscope-app',
+        title: 'Open in Profiles Drilldown',
+        onClick: jest.fn(),
+      }]
+    });
+
+    const result = getSpanDetailLinkButtons({
+      span,
+      createSpanLink,
+      datasourceType: 'test',
+      traceToProfilesOptions: {
+        datasourceUid: 'test-uid',
+        profileTypeId: 'test-type',
+        customQuery: false,
+      },
+      timeRange,
+      app: CoreApp.Explore
+    });
+
+    expect(result.profileLinkButtons).toBeDefined();
+    // Should render both the original profile link and the drilldown button
+    expect(result.profileLinkButtons?.props.children).toHaveLength(2);
+  });
+
+  it('should not create profile drilldown button when not in Explore', () => {
+    createSpanLink.mockReturnValue([{ type: SpanLinkType.Profiles, href: '/profiles', title: RelatedProfilesTitle }]);
+    (usePluginLinks as jest.Mock).mockReturnValue({
+      isLoading: false,
+      links: [{
+        pluginId: 'grafana-pyroscope-app',
+        title: 'Open in Profiles Drilldown',
+        onClick: jest.fn(),
+      }]
+    });
+
+    const result = getSpanDetailLinkButtons({
+      span,
+      createSpanLink,
+      datasourceType: 'test',
+      traceToProfilesOptions: {
+        datasourceUid: 'test-uid',
+        profileTypeId: 'test-type',
+        customQuery: false,
+      },
+      timeRange,
+      app: CoreApp.Dashboard
+    });
+
+    expect(result.profileLinkButtons).toBeDefined();
+    // Should only render the original profile link
+    expect(result.profileLinkButtons?.props.children).toBeFalsy();
```

Comment:
Why it's `.toBeFalsy();` if one link is expected? Shouldn't it be `.toHaveLength(1);`?

Reply:
In this case we're not returning two children i.e. two buttons wrapped in a container. We're only returning one button and no children.